controllers/webhook/webhook.go

```diff
@@ -36,25 +37,29 @@ var log = ctrl.Log.WithName("odh-controller-webhook")
 //nolint:lll
 
 type OpenDataHubWebhook struct {
-	client  client.Client
-	decoder *admission.Decoder
+	Client  client.Client
+	Decoder *admission.Decoder
 }
 
-func (w *OpenDataHubWebhook) SetupWithManager(mgr ctrl.Manager) {
+func (w *OpenDataHubWebhook) SetupWithManager(mgr ctrl.Manager) error {
+	// Initialize
+	if mgr == nil {
+		return errors.New("operator manager is nil, should init first")
+	}
+	if w.Client == nil { // replace the old InjectClient() method
```

Comment:
It is ont time initialization call from main so the nil checks are probably redundant, they are guaranteed nil.

Reply:
maybe better to get w.Decoder = admission.NewDecoder(mgr.GetScheme()) set in main as well.
so i

Reply:
Wait a moment. Client actually set in main. I see 2 options:
1) do like with the reconcilers in main, set both Client and Decoder there.
2) change it to not create OpenDataHubWebhook in the main at all but hide all the machinery under some call SetupWebhook(mgr)